solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -50,15 +49,42 @@ public class SizeAwareDirectory extends FilterDirectory
   private boolean initialized = false;
   private volatile long reconciledTimeNanos;
   private volatile LongAdder size = new LongAdder();
-  private volatile LongObjectProcedure<String> sizeWriter = (size, name) -> this.size.add(size);
+  private volatile LongAdder onDiskSize = new LongAdder();
+  private volatile SizeWriter sizeWriter =
+      (size, onDiskSize, name) -> {
+        this.size.add(size);
+        this.onDiskSize.add(onDiskSize);
+      };
 
-  private final ConcurrentHashMap<String, Long> fileSizeMap = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, Sizes> fileSizeMap = new ConcurrentHashMap<>();
 
   private final ConcurrentHashMap<String, SizeAccountingIndexOutput> liveOutputs =
       new ConcurrentHashMap<>();
 
   @SuppressWarnings({"unchecked", "rawtypes"})
-  private final Future<Long>[] computingSize = new Future[1];
+  private final Future<Sizes>[] computingSize = new Future[1];
+
+  private interface SizeWriter {
+    void apply(long size, long onDiskSize, String name);
+  }
+
+  private static class Sizes implements Accountable {
+    private static final long RAM_BYTES_USED =
+        RamUsageEstimator.shallowSizeOfInstance(SizeAccountingIndexOutput.class);
```

Comment:
This should instead be `RamUsageEstimator.shallowSizeOfInstance(SizeAccountingIndexOutput.Sizes.class)`, no?